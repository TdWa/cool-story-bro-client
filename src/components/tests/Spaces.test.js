import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import Space from "../Space";

const props = {
  id: 99,
  backgroundColor: "white",
  color: "black",
  title: "AwesomeSpace",
  description: "This is MY space",
  showLink: true,
};

describe("<Space />", () => {
  test.skip("should render without props", () => {
    // Should it? this is a decision you should make over your components.
    // Writing tests helps us think of all the open possibilities we left
    // in our code and how they would affect our app.

    const component = renderer.create(<Space />); // "render" the component
    const tree = component.toJSON(); // Convert it to JSON
    expect(tree).toMatchSnapshot(); // Compare to stored snapshot
  });

  test.skip("should render with all props", () => {
    const component = renderer.create(<Space {...props} />); // "render" the component
    const tree = component.toJSON(); // Convert it to JSON
    expect(tree).toMatchSnapshot(); // Compare to stored snapshot
  });

  test("A Link should render when showLink prop is set to true", () => {
    const component = renderer.create(
      <MemoryRouter>
        <Space {...props} />
      </MemoryRouter>
    ); // "render" the component
    const tree = component.toJSON(); // Convert it to JSON
    expect(tree).toMatchSnapshot(); // Compare to stored snapshot
  });
});
