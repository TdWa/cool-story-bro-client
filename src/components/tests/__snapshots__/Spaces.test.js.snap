// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Space /> A Link should render when showLink prop is set to true 1`] = `
<div
  className="jumbotron"
  style={
    Object {
      "backgroundColor": "white",
      "color": "black",
    }
  }
>
  <h1>
    AwesomeSpace
  </h1>
  <p>
    This is MY space
  </p>
  <a
    href="/spaces/99"
    onClick={[Function]}
  >
    <button
      className="btn btn-primary"
      disabled={false}
      type="button"
    >
      Visit space
    </button>
  </a>
</div>
`;

exports[`<Space /> should render with all props 1`] = `
<div
  className="jumbotron"
  style={
    Object {
      "backgroundColor": "white",
      "color": "black",
    }
  }
>
  <h1>
    AwesomeSpace
  </h1>
  <p>
    This is MY space
  </p>
</div>
`;

exports[`<Space /> should render with all props 2`] = `
<div
  className="jumbotron"
  style={
    Object {
      "backgroundColor": "white",
      "color": "black",
    }
  }
>
  <h1>
    AwesomeSpace
  </h1>
  <p>
    This is MY space
  </p>
</div>
`;

exports[`<Space /> should render with props 1`] = `
<div
  className="jumbotron"
  style={
    Object {
      "backgroundColor": "white",
      "color": "black",
    }
  }
>
  <h1>
    AwesomeSpace
  </h1>
  <p>
    This is MY space
  </p>
</div>
`;

exports[`<Space /> should render without props 1`] = `
<div
  className="jumbotron"
  style={
    Object {
      "backgroundColor": undefined,
      "color": undefined,
    }
  }
>
  <h1 />
  <p />
</div>
`;

exports[`<Space /> should render without props 2`] = `
<div
  className="jumbotron"
  style={
    Object {
      "backgroundColor": undefined,
      "color": undefined,
    }
  }
>
  <h1 />
  <p />
</div>
`;
